web.ml

```diff
@@ -242,13 +242,20 @@ module Http (IO : IO_TYPE) (Curl_IO : CURL with type 'a t = 'a IO.t) : HTTP with
     | _ ->
       bprintf b "error (%d) %s (errno %d)" (errno code) (strerror code) (Curl.get_oserrno h)
     end;
-    log #info_s (Buffer.contents b);
-    return ()
+    log #info_s (Buffer.contents b)
 
   (* NOTE don't forget to set http_1_0=true when sending requests to a Httpev-based server *)
   (* Don't use curl_setheaders when using ?headers option *)
   let http_request' ?ua ?timeout ?(verbose=false) ?(setup=ignore) ?timer ?max_size ?(http_1_0=false) ?headers ?body (action:http_action) url =
     let open Curl in
+    let action_name = string_of_http_action action in
+
+    let headers = match Possibly_otel.get_traceparent () with
+    | None -> headers
+    | Some tp_header ->
+      Some (tp_header :: (Option.default [] headers))
```

Comment:
how do we know tp_header is "safe" - is it guaranteed that it is a valid HTTP header and that it doesn't overlap with any other header passed in? Or it is claimed to be caller responsibility? Then should be documented?

Reply:
The implementation of `get_traceparent` is here in the PR;

https://github.com/ahrefs/devkit/pull/19/files/6e30bb59b33a9e1072f301de5cc339a1a3fdec27#diff-0a4a651748420c02be78207e130336017e1cc582920c94cae789ca8609189d3fR3-R10

… which calls [`get_ambient_scope`](https://github.com/imandra-ai/ocaml-opentelemetry/blob/e53cffbf12dde76e53eb5195b228001b5010b728/src/core/opentelemetry.ml#L601-L605) from `ocaml-opentelemetry`; which isn't _that_ well documented, I suppose, heh; but it's an opaque that can only be set with `with_ambient_scope` in a type-safe way (i.e. it's being set with a structured record of `Otel.Scope.{ trace_id; span_id; _ }`, not a string; and both of those are also opaque types with constructors, etc ...)

Beyond this all being typechecked, though, I'm not sure how to make this safer, unless you have a specific suggestion? 😄